src/Service.Tests/Configuration/ConfigurationTests.cs

```diff
@@ -1331,6 +1331,177 @@ public async Task TestRuntimeBaseRouteInNextLinkForPaginatedRestResponse()
             }
         }
 
+        /// <summary>
+        /// Validates the Location header field returned as part of POST requests.
+        /// </summary>
+        /// <param name="entityType">Type of the entity</param>
+        /// <param name="requestPath">Request path for performing POST API requests on the entity</param>
+        /// <returns></returns>
+        [DataTestMethod]
+        [TestCategory(TestCategory.MSSQL)]
+        [DataRow(EntitySourceType.Table, "/api/Book", DisplayName = "Location Header - Tables, Base Route not configured")]
+        [DataRow(EntitySourceType.StoredProcedure, "/api/GetBooks", DisplayName = "Location Header - Stored Procedures, Base Route not configured")]
+        public async Task ValidateLocationHeaderFieldForPostRequests(EntitySourceType entityType, string requestPath)
+        {
+
+            GraphQLRuntimeOptions graphqlOptions = new(Enabled: true);
+            RestRuntimeOptions restRuntimeOptions = new(Enabled: true);
+
+            DataSource dataSource = new(DatabaseType.MSSQL,
+                GetConnectionStringFromEnvironmentConfig(environment: TestCategory.MSSQL), Options: null);
+
+            RuntimeConfig configuration;
+
+            if (entityType is EntitySourceType.StoredProcedure)
+            {
+                Entity entity = new(Source: new("get_books", EntitySourceType.StoredProcedure, null, null),
+                              Rest: new(new SupportedHttpVerb[] { SupportedHttpVerb.Get, SupportedHttpVerb.Post }),
+                              GraphQL: new(Singular: "GetBooks", Plural: "GetBooks"),
+                              Permissions: new[] { GetMinimalPermissionConfig(AuthorizationResolver.ROLE_ANONYMOUS) },
+                              Relationships: null,
+                              Mappings: null
+                             );
+
+                string entityName = "GetBooks";
+                configuration = InitMinimalRuntimeConfig(dataSource, graphqlOptions, restRuntimeOptions, entity, entityName);
+            }
+            else
+            {
+                configuration = InitMinimalRuntimeConfig(dataSource, graphqlOptions, restRuntimeOptions);
+            }
+
+            const string CUSTOM_CONFIG = "custom-config.json";
+            File.WriteAllText(CUSTOM_CONFIG, configuration.ToJson());
+            string[] args = new[]
+            {
+                $"--ConfigFileName={CUSTOM_CONFIG}"
+            };
+
+            using (TestServer server = new(Program.CreateWebHostBuilder(args)))
+            using (HttpClient client = server.CreateClient())
+            {
+                HttpMethod httpMethod = SqlTestHelper.ConvertRestMethodToHttpMethod(SupportedHttpVerb.Post);
+                HttpRequestMessage request = new(httpMethod, requestPath);
```

Comment:
For tests performing mutations, we reset the database state after performing the mutation operation. However, here it would be difficult to reset the state unless we do something like we do with RestApiTestBase. You might consider performing a subsequent delete operation after performing the GET operation in this test. This way, the test would not change the state of the database.

Reply:
I touched upon this in an earlier comment also. This was one of the reasons for me to choose an entity whose PK is auto generated. As part of this test, a single new book item is created whose id cannot be known beforehand. So, we can be sure that the generated id for this item could not have been used in any other test for read, update, delete etc. AFAIK, all the integration tests (except for the ones in ConfigurationTests) has a test setup phase where the database state is reset to scratch by executing the .sql scripts. Creation of two new items in this entity shouldn't interfere with other tests.

Reply:
I see. So unless I run other tests which reset the database, this tests will continue to insert new records everytime it is run?

Reply:
Thanks for the suggestion. I've added a DELETE request to cleanup the newly created item